File: DesktopSample/DataBinding/PageAndRDL/XmlDataSource/VB.NET/XmlDataSource/DataLayer.vb
Imports System.Xml
Imports System.Xml.XPath

' サンプルで使用するデータを提供します。
Friend NotInheritable Class DataLayer

	Public Function CreateReader() As XmlReader
		Dim txtReader As New XmlTextReader("..\..\..\MyXmlDB.xml")
		Return txtReader
	End Function

	Public Function CreateDocument() As IXPathNavigable
		Dim doc As New XPathDocument("..\..\..\MyXmlDB.xml")
		Return doc
	End Function

End Class

File: DesktopSample/DataBinding/PageAndRDL/XmlDataSource/C#/XmlDataSource/DataLayer.cs
using System.Xml;
using System.Xml.XPath;

namespace ActiveReports.Samples.XmlDataSource
{
	// サンプルで使用するデータを提供します。
	internal sealed class DataLayer
	{
		public XmlReader CreateReader()
		{
			var txtReader = new XmlTextReader(@"..\..\..\MyXmlDB.xml");
			return txtReader;
		}

		public IXPathNavigable CreateDocument()
		{
			var doc = new XPathDocument(@"..\..\..\MyXmlDB.xml");
			return doc;
		}
	}
}
